### PR TITLE
[MAINTENANCE] Raise error on unsupported versions of python.

### PR DESCRIPTION
**Pull request by billdirks**
```
% python                 
Python 3.11.3 (main, May 15 2023, 10:43:03) [Clang 14.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import great_expectations
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/bill/311Test/great_expectations/great_expectations/__init__.py", line 9, in <module>
    raise ImportError(
ImportError: Great Expectations is only supported on python 3.8 through 3.10. You are using: 3.11.3 (main, May 15 2023, 10:43:03) [Clang 14.0.6 ]
>>> 
```

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/__init__.py [hunk 1]
```diff
@@ -1,3 +1,17 @@
+import sys
+
+PYTHON_SUPPORTED_VERSION = {"major": 3, "minor": {"lower": 8, "upper": 10}}
```
Comment (cdkini) [inline great_expectations/__init__.py]: we could use the `version` package here to make this easier.

`version.Version` has built in comparators

File: great_expectations/__init__.py [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+if (sys.version_info.major != PYTHON_SUPPORTED_VERSION["major"]) or (
+    sys.version_info.major == PYTHON_SUPPORTED_VERSION["major"]
+    and sys.version_info.minor < PYTHON_SUPPORTED_VERSION["minor"]["lower"]
+    or sys.version_info.minor > PYTHON_SUPPORTED_VERSION["minor"]["upper"]
+):
+    raise ImportError(
+        "Great Expectations is only supported on python 3.8 through 3.10. "
```
Comment (cdkini) [inline great_expectations/__init__.py]: Nit - Could we capitalize Python?

File: great_expectations/__init__.py (hunk 1 above)
Comment (cdkini) [reply great_expectations/__init__.py]: `from packaging import version` I believe

File: great_expectations/__init__.py (hunk 2 above)
Comment (billdirks) [reply great_expectations/__init__.py]: Done.

File: great_expectations/__init__.py (hunk 1 above)
Comment (billdirks) [reply great_expectations/__init__.py]: I would prefer not to take on a dependency. `sys` is builtin and the conditional is not complicated.
I do see comparators builtin if you pass in a string representation to `version.Version` but I don't know how to get this representation of the python runtime without using `sys` and then doing some manipulation of `sys.version` or `sys.version_info` which seems more complicated than this.

**Review by Shinnnyshinshin — APPROVED** [review]
I was really looking forward to this :) Thank you @billdirks

File: great_expectations/__init__.py [hunk 3]
```diff
@@ -1,3 +1,17 @@
+import sys
+
+PYTHON_SUPPORTED_VERSION = {"major": 3, "minor": {"lower": 8, "upper": 10}}
+if (sys.version_info.major != PYTHON_SUPPORTED_VERSION["major"]) or (
+    sys.version_info.major == PYTHON_SUPPORTED_VERSION["major"]
```
Comment (Kilo59) [inline great_expectations/__init__.py]: Why do we care about major version at all?
Python 4 isn't going to be a thing, and users can't install us with Python 2?

Comment (billdirks) [reply great_expectations/__init__.py]: I guess I like to be explicit. People could install use via a `git clone` and not be gated via `pip`. I don't think python 2 users are going to use us but my preference is to leave the check in.

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**